taxi/views.py

```diff
@@ -1,47 +1,54 @@
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.shortcuts import render
 from django.views import generic
 
 from .models import Driver, Car, Manufacturer
 
 
+@login_required
 def index(request):
     """View function for the home page of the site."""
 
     num_drivers = Driver.objects.count()
     num_cars = Car.objects.count()
     num_manufacturers = Manufacturer.objects.count()
 
+    num_visits = request.session.get("num_visits", 0)
+    request.session["num_visits"] = num_visits + 1
+
     context = {
         "num_drivers": num_drivers,
         "num_cars": num_cars,
         "num_manufacturers": num_manufacturers,
+        "num_visits": num_visits + 1
     }
```

Comment:
Did you have "num_visit" == 1?
Looks like you  make +1 two times

Reply:
I separately iterate through request.session["num_visits"] and display the pre-assigned value num_visits + 1 :)